### PR TITLE
Use Bootstrap form builder

### PR DESCRIPTION
**Pull request by chrisma**

### CODE REVIEW

**Comment by BjoernDaase** [comment]
@chrisma Do you also plan to adjust the files where we currently use our custom `HARTFormBuilder`.
The reason why I ask is, that I do not see a design at the moment, how we fit the 
Screenshot from 2019-03-28 18-55-03
style in e.g. the already halved pages like 
Screenshot from 2019-03-28 19-00-44

**Comment by chrisma** [comment]
> I do not see a design at the moment, how we fit the style in e.g. the already halved pages like

@BjoernDaase Thanks for checking out the code. The horizontal layout seen in some of the new forms is a helper of the `bootstrap_form` gem that allows easy layouting (https://github.com/bootstrap-ruby/bootstrap_form#horizontal-forms). It is a single setting on the form object: `layout: :horizontal`.

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -1,67 +1,72 @@
...
   def authenticate_responsible_user
     @project = Project.find(params[:id])
     redirect_to dashboard_path, alert: I18n.t('authorization.unauthorized') unless @project.responsible_users.include? current_user
   end
 
   def project_params
-    params.require(:project).permit(:name, :description, responsible_user_ids: [])
+    p = params.require(:project).permit(:name, :description, responsible_users_ids: [])
+    # The projects form returns a list of ids of responsible_users, turn them into user objects
+    responsible_users = p.delete(:responsible_users_ids).map { |id| User.find_by_id(id) }.reject(&:nil?)
```
Comment (chrisma) [inline app/controllers/projects_controller.rb]: Is there a method for getting Users for a list of ids? Surely, that's a thing

Comment (chrisma) [reply app/controllers/projects_controller.rb]: `User.where` deals nicely with lists of ids without throwing `RecordNotFound` execeptions

File: app/views/operating_systems/_form.html.erb [hunk 2]
```diff
@@ -1,22 +1,6 @@
...
-  </div>
-
-  <div class="actions mt-3">
-    <%= form.submit class: "btn btn-primary" %>
-  </div>
+<%# https://github.com/bootstrap-ruby/bootstrap_form %>
+<%= bootstrap_form_with(model: operating_system, local: true) do |form| %>
+  <%= form.alert_message "Please fix the errors below.", error_summary: false %>
+  <%= form.text_field :name %>
+  <%= form.submit class: "btn btn-primary" %>
```
Comment (chrisma) [inline app/views/operating_systems/_form.html.erb]: replace with `form.primary`

File: app/views/requests/_form.html.erb [hunk 3]
```diff
@@ -1,89 +1,57 @@
...
-                          {class: "form-control"} %>
-        </div>
-        <div class="form-group row full-width pt-3">
-          <div class="col">
-            <%= form.label :cpu_cores, "CPU cores *" %>
-            <%= form.number_field :cpu_cores, min: 0, class: 'form-control', id: 'cpu' %>
+        <%= form.select :template_id,
+                        options_for_select(@request_templates.map {|request_template|
+                          ["#{request_template.name}: #{request_template.cpu_cores} CPU cores, #{request_template.ram_gb} GB RAM, #{request_template.storage_gb} GB Storage, #{request_template.operating_system}",
+                          "[#{request_template.cpu_cores}, #{request_template.ram_gb}, #{request_template.storage_gb}, '#{request_template.operating_system}']"]}),
```
Comment (chrisma) [inline app/views/requests/_form.html.erb]: This formatting could be a method on the `RequestTemplate` model

File: app/views/requests/_form.html.erb [hunk 4]
```diff
@@ -1,89 +1,57 @@
...
+          <div class="col-lg-4">
+            <%= form.number_field :storage_gb, min: 0, id: 'storage', label: 'Storage in GB *' %>
           </div>
         </div>
-        <div class="form-group pt-3">
-          <%= form.label :operating_system, "Operating System *" %>
-          <%= form.select :operating_system, operating_system_options, {}, id: 'operating_system', class: 'form-control' %>
-        </div>
+        <%= form.select :operating_system, 
+          operating_system_options,
```
Comment (chrisma) [inline app/views/requests/_form.html.erb]: what are these?

File: app/views/servers/_form.html.erb [hunk 5]
```diff
@@ -1,85 +1,31 @@
...
-                    options_from_collection_for_select(User.all,
-                                                       :id,
-                                                       :human_readable_identifier),
-                    {},
-                    { multiple: false, class: "selecttwo", style: 'width: 50%' } %>
-  </div>
-
+<%# https://github.com/bootstrap-ruby/bootstrap_form %>
+<%= bootstrap_form_with(model: server, local: true, layout: :horizontal) do |form| %>
+  <%= form.alert_message "Please fix the errors below.", error_summary: false %>
```
Comment (chrisma) [inline app/views/servers/_form.html.erb]: Comment for documentation on `alert_message`

File: app/views/vms/edit_config.html.erb [hunk 6]
```diff
@@ -1,56 +1,35 @@
...
-    <%= form.labeled_text_field 'DNS', :dns %>
-  </div>
+<h5>Network Adapters</h5>
+<dl>
+  <% @vm.macs.each do |adapter_name, mac_address| %>
+    <dt><%= adapter_name %></dt>
+    <dd>
+      <%= mac_address %>
+      <span
+        class="btn btn-outline-primary btn-sm fa fa-copy clipboard-btn"
```
Comment (chrisma) [inline app/views/vms/edit_config.html.erb]: Readd comment `<%# Copy to clipboard button, JS matches on 'clipboard-btn' class %>`

File: app/views/requests/_form.html.erb (hunk 3 above)
Comment (chrisma) [reply app/views/requests/_form.html.erb]: `RequestTemplate#text_summary`. Also uses JSON in the option values.
https://github.com/hpi-swt2/vm-portal/pull/525/commits/e7f21d988e2d5a146d2a66e82e6ff3e20a855e9e

**Comment by chrisma** [comment]
@BjoernDaase @MrModder @BraunTom This PR is ready for review :)

**Pull request merged**